### ISSUE TITLE
Add option to limit download speed when writing stream data to file

### ISSUE THREAD

**Issue by mepherion** [streamlink/streamlink#6065]
### Checklist

- [X] This is a feature request and not [a different kind of issue](https://github.com/streamlink/streamlink/issues/new/choose)
- [X] [I have read the contribution guidelines](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#contributing-to-streamlink)
- [X] [I have checked the list of open and recently closed feature requests](https://github.com/streamlink/streamlink/issues?q=is%3Aissue+label%3A%22feature+request%22)

### Collaboration

- [X] My request is genuine and in the best interest of the project
- [X] [I will provide feedback should a pull request be opened with a new feature implementation](https://github.com/streamlink/streamlink/blob/master/CONTRIBUTING.md#pull-request-feedback)

### Description

Suggesting an ability for streamlink to limit the speed of downloading the stream data when writing to a file to prevent oversaturating the network connection, similar to the `--limit-rate` option from `yt-dlp`

For example: `streamlink --output FILENAME --limit-rate RATE ...` where `RATE` is the maximum download rate in bytes per second, e.g. `50K` or `4.2M`

**Comment by bastimeyer** [comment]
Going to close this, because Streamlink's focus is on live streaming, and not on downloading VOD content. Throttling download speeds is irrelevant for live streams.

yt-dlp implements download throttling in a single downloader/file-writer thread ([1](https://github.com/yt-dlp/yt-dlp/blob/6075a029dba70a89675ae1250e7cdfd91f0eba41/yt_dlp/downloader/http.py#L201-L365), [2](https://github.com/yt-dlp/yt-dlp/blob/6075a029dba70a89675ae1250e7cdfd91f0eba41/yt_dlp/downloader/common.py#L203-L217)), where this is trivially implemented using `time.sleep()`.

Streamlink on the other hand implements segmented stream types (HLS / DASH) using a [`ThreadPool`](https://github.com/streamlink/streamlink/blob/6.8.2/src/streamlink/stream/segmented/segmented.py#L73) of multiple download threads which write into a common [`RingBuffer`](https://github.com/streamlink/streamlink/blob/6.8.2/src/streamlink/buffers.py#L74-L160) whose content is then [read and written](https://github.com/streamlink/streamlink/blob/6.8.2/src/streamlink_cli/streamrunner.py#L119-L134) to the output. Applying throttling to the output doesn't apply throttling to the segmented thread-pool threads. On top of that, each HTTP response must be read in streaming mode ([`--hls-segment-stream-data`](https://streamlink.github.io/cli.html#cmdoption-hls-segment-stream-data)) in order to be ableto control the download speed, which isn't the default behavior. So by default, entire segments are buffered as a whole by Streamlink's `requests` dependency before the content is written to Streamlink's output buffer. Writing whole segment contents is also necessary for encrypted HLS streams because of the unpadding that needs to be done, so throttling is impossible here. And compared to HLS streams, DASH streams currently lack such streaming-downloading option.

TL;DR
Not worth the effort. If the lack of throttling bothers you, then use a different tool for downloading VOD content.
